extensions/amp-next-page/1.0/service.js

```diff
@@ -170,43 +162,48 @@ export class NextPageService {
     // Save the <amp-next-page> from the host page
     this.host_ = element;
 
+    // Parse attributes
+    this.nextSrc_ = this.getHost_().getAttribute('src');
+    this.hasDeepParsing_ =
+      this.getHost_().hasAttribute('deep-parsing') || !this.nextSrc_;
```

Comment:
What's deep parsing?

Reply:
See https://github.com/ampproject/amphtml/blob/b449a4ceee39f31ab7b233751531c1a0369b501a/extensions/amp-next-page/amp-next-page.md#deep-parsing-optional

Reply:
Why is the flag needed? Can't we enforce the JSON response to be flat? I'm a bit confused about the use-case or the performance concern. Sorry for missing this in previous reviews.

(Also non-blocking for this PR btw.)

Reply:
The flag is needed in the particular case where the user wants to limit the loading of articles to an inlined list of pages even though those pages themselves have `<amp-next-page>` elements inside of them (whereas the default behavior is to recursively parse pages from the inline config of each of the sub-articles).